src/Dictionaries/MySQLDictionarySource.cpp

```diff
@@ -114,13 +121,41 @@ std::string MySQLDictionarySource::getUpdateFieldAndDate()
     }
 }
 
+BlockInputStreamPtr MySQLDictionarySource::retriedCreateMySqlBIStream(const std::string & data_fetch_query_str, const size_t max_tries)
+{
+    size_t count_connection_lost = 0;
+
+    while (true)
+    {
+        auto connection = pool.get();
+
+        try
+        {
+            return std::make_shared<MySQLBlockInputStream>(
+                    connection, data_fetch_query_str, sample_block, max_block_size, close_connection);
+        }
+        catch (const mysqlxx::ConnectionLost & ecl)  /// There are two retriable failures: CR_SERVER_GONE_ERROR, CR_SERVER_LOST
+        {
+            if (++count_connection_lost < max_tries)
+            {
+                LOG_WARNING(log, ecl.displayText());
+                LOG_WARNING(log, "No connection. Trying to reconnect...");
```

Comment:
`No connection` -> `Cannot connect to MySQL server`? (or something more readable)
Also add count_connection_lost, max_tries to message

Reply:
`No connection` is actually more meaningful here, because it has been lost due to some reason - `catch (const mysqlxx::ConnectionLost & ecl)`